File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.c
int minimumArea(int** grid, int gridSize, int* gridColSize) {
    int min_x = 1001;
    int max_x = -1;
    int min_y = 1001;
    int max_y = -1;

    for (int x = 0; x < gridSize; x++) {
        for (int y = 0; y < *gridColSize; y++) {
            if (grid[x][y]) {
                if (x < min_x) min_x = x;
                if (x > max_x) max_x = x;
                if (y < min_y) min_y = y;
                if (y > max_y) max_y = y;
            }
        }
    };

    return (max_x - min_x + 1) * (max_y - min_y + 1);
}

File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        min_x, max_x, min_y, max_y = 1001, -1, 1001, -1

        for x in range(len(grid)):
            for y in range(len(grid[0])):
                g = grid[x][y]
                if g:
                    min_x = min(min_x, x)
                    max_x = max(max_x, x)
                    min_y = min(min_y, y)
                    max_y = max(max_y, y)
     
        return (max_x - min_x + 1) * (max_y - min_y + 1)